3/apollo.js

```diff
@@ -20,85 +18,156 @@ const typeDefs = gql`
-      { id: '2', message: 'Bar', assignee: assignees[1],},
-      { id: '3', message: 'Baz', assignee: assignees[0],}
-    ];
-
 function getRandomId(){
 	return Math.floor(Math.random() * Math.floor(9999999)).toString();
 }	
 	
 const resolvers = {
   Query: {
 	todo: (parent, args, context, info) => {
-		return todos.find((todo)=>{return todo.id == args.id});
+		return neo4jgraphql(parent, args, context, info)
 	},
-    todos: (parent, args, context, info) => {
-		let result = todos;
+    todos: async(parent, args, context, info) => {
+		const session = context.driver.session()
+		let result
 		if(args.assignee != null){
-			result = todos.filter((todo)=>{return todo.assignee.id == args.assignee});
+			let query = `
+			MATCH (t:Todo)-[:IS_ASSIGNED_TO]->(p:Person {id: $assignee})
+			RETURN t, p
+			ORDER BY p.name
+			SKIP $s
+			LIMIT $l
+			`
+			if(args.desc == true){
+				query = `
+				MATCH (t:Todo)-[:IS_ASSIGNED_TO]->(p:Person {id: $assignee})
+				RETURN t, p
+				ORDER BY p.name DESC
+				SKIP $s
+				LIMIT $l
+				`
+			}
+			result = await session.run(query,
+			{
+				assignee: args.assignee,
+				s: args.offset,
+				l: args.first
+			})
+		}else{
+			let query = `
+			MATCH (t:Todo)-[:IS_ASSIGNED_TO]->(p:Person)
+			RETURN t, p
+			ORDER BY p.name
+			SKIP $s
+			LIMIT $l
+			`
+			if(args.desc == true){
+				query = `
+				MATCH (t:Todo)-[:IS_ASSIGNED_TO]->(p:Person)
+				RETURN t, p
+				ORDER BY p.name DESC
+				SKIP $s
+				LIMIT $l
+				`
+			}
+			result = await session.run(query,
+			{
+				s: args.offset,
+				l: args.first
+			})			
 		}
-		return result;
+
+		session.close()
+		return result.records.map((record) => {
+			let response = {...record.get('t').properties, assignee: record.get('p').properties}
+			return response;
+		});
 	},
   },
   Mutation: {
-	  login: (parent, args, context, info) =>{
-		  return jwt.sign({userName: args.userName}, secret, {expiresIn: "1 day"});
+	  login: async(parent, args, context, info) =>{
+		  const session = context.driver.session()
+		  const result = await session.run(`
+			  MATCH (p:Person {name: $name, password: $password})
+			  RETURN p
+			  `,
+		  {
+			name: args.userName,
+			password: args.pwd
+		  }
+		  )
+		  const record = result.records[0];
+		  if(record){
+			  return jwt.sign({userName: args.userName, }, secret, {expiresIn: "1 day"});
+		  }else{
+			  return ""
+		  }
 	  },
-	  createTodo: (parent, args, context, info) =>{
-		  todos.push({
-			  id: getRandomId(),
-			  message: args.message,
-			  assignee: assignees.find((assignee)=>{ return assignee.id == args.assignee})
-		  });
-		  return todos[todos.length -1];
+	  createTodo: async (parent, args, context, info) => {
+		  const session = context.driver.session()
+		  const result = await session.run(`
+		  MATCH (p:Person) WHERE p.id = $assignee
+		  CREATE (t:Todo {id: $id, message: $message})
+		  MERGE (t)-[:IS_ASSIGNED_TO]->(p)
+		  RETURN t, p
```

Comment:
Not sure what will happen when assignee ID, it's to say, person's ID is not in the database, e.g. `id = 123156576313543`. `MERGE` functions as upsert, so if the user is `null`, it will annul the entire creation of the new todo, instead of creating a new one with `assignee = null`, due to `MATCH` returning `null`. Interesting, except from when the logic behind is: a todo could only be created if it's assigned to some person by the time of its creation.